proxy/src/lib.rs

```diff
@@ -317,8 +318,23 @@ where
         // Clone the router handle
         let router = router.clone();
 
-        // Map errors to 500 responses
-        MapErr::new(router)
+        // Map errors to appropriate response error codes.
+        MapErr::new(router, |e| {
+            match e {
+                RouteError::Route(r) => {
+                    error!("route error: {}", r);
+                    http::StatusCode::INTERNAL_SERVER_ERROR
+                }
+                RouteError::Inner(i) => {
+                    error!("{}", i);
+                    http::StatusCode::INTERNAL_SERVER_ERROR
+                }
+                RouteError::NotRecognized => {
+                    error!("route not recognized");
+                    http::StatusCode::NOT_FOUND
```

Comment:
What instances does this `NotRecognized` error occur? Is it true that it is **always** the user's fault?

Reply:
Hmm, I'll have to take a closer look at that.

Reply:
I believe that `NotRecognized` is returned in the following cases:

+ By the outbound side of the proxy, if the request had no authority, no host header, and no SO_ORIGINAL_DST (or, we were on an OS that doesn't support SO_ORIGINAL_DST):
https://github.com/runconduit/conduit/blob/82bbcbd13769227aac7a1354b4106d6818f671e7/proxy/src/outbound.rs#L83-L85

+ By the inbound side of the proxy if the request had no SO_ORIGINAL_DST and the inbound was not configured with a default address.

I'm not sure if 404 is the correct error code to return in these cases or not --- what do you think? Are these bad requests?

Reply:
@seanmonstar as of 06a7b8f3be590347c4bda8514b6db208b3062ee9, everything gets 500 errors once again. Would still like to return more accurate error codes in a future PR, but this is the most correct behavior for now, and I'd like to get the improved logging merged.